File: GeeksforGeeks/Hashing/count_k_diff_pair.java
//https://practice.geeksforgeeks.org/problems/count-distinct-pairs-with-difference-k1233/1#

class Solution
{
    public int TotalPairs(int[] nums, int k)
    {
        // Code here
        int ans = 0;
        
        HashMap<Integer, Integer> mp = new HashMap<>();
        
        // create hashmap
        for(int itr : nums) {
            
            if(mp.containsKey(itr))
            mp.put(itr,mp.get(itr)+1);
            else
            mp.put(itr,1);
        }
        
        for(int itr : mp.keySet()){
            
            if(k == 0){
                
                if(mp.get(itr) > 1)         ans++;
            }
            else{
                
                if(mp.containsKey(itr+k))   ans++;
            }
        }
        
        return ans;
    }
}

File: GeeksforGeeks/Hashing/count_k_diff_pair.cpp
//https://practice.geeksforgeeks.org/problems/count-distinct-pairs-with-difference-k1233/1#

class Solution{
    public:
    int TotalPairs(vector<int>nums, int k)
    {
        // Code here
        
        int ans = 0;
        unordered_map<int,int> ump;
        
        for(auto itr: nums) ump[itr]++;
        
        for(auto itr: ump){
            
            int key = itr.first;
            int freq = itr.second;
            
            if(k == 0)
            {
                if(freq > 1)    ans++;
            }
            else
            {
                if(ump.count(key+k))    ans++;
            }
        }
        
        return ans;
    }

};